diff --git a/vendor/knative.dev/serving/pkg/reconciler/testing/v1alpha1/factory.go b/vendor/knative.dev/serving/pkg/reconciler/testing/v1alpha1/factory.go
index 0af207dc..92129722 100644
--- a/vendor/knative.dev/serving/pkg/reconciler/testing/v1alpha1/factory.go
+++ b/vendor/knative.dev/serving/pkg/reconciler/testing/v1alpha1/factory.go
@@ -21,6 +21,7 @@ import (
 	"encoding/json"
 	"testing"
 
+	fakesmmrclient "github.com/openshift-knative/knative-serving-networking-openshift/pkg/client/maistra/injection/client/fake"
 	fakerouteclient "github.com/openshift-knative/knative-serving-networking-openshift/pkg/client/openshift/injection/client/fake"
 	fakecachingclient "knative.dev/caching/pkg/client/injection/client/fake"
 	fakesharedclient "knative.dev/pkg/client/injection/client/fake"
@@ -76,6 +77,7 @@ func MakeFactory(ctor Ctor) rtesting.Factory {
 		ctx, cachingClient := fakecachingclient.With(ctx, ls.GetCachingObjects()...)
 		ctx, certManagerClient := fakecertmanagerclient.With(ctx, ls.GetCMCertificateObjects()...)
 		ctx, routeClient := fakerouteclient.With(ctx, ls.GetOpenshiftObjects()...)
+		ctx, smmrClient := fakesmmrclient.With(ctx, ls.GetServiceMeshMemberRollObjects()...)
 		ctx = context.WithValue(ctx, TrackerKey, &rtesting.FakeTracker{})
 
 		// The dynamic client's support for patching is BS.  Implement it
@@ -117,6 +119,7 @@ func MakeFactory(ctor Ctor) rtesting.Factory {
 			cachingClient.PrependReactor("*", "*", reactor)
 			certManagerClient.PrependReactor("*", "*", reactor)
 			routeClient.PrependReactor("*", "*", reactor)
+			smmrClient.PrependReactor("*", "*", reactor)
 		}
 
 		// Validate all Create operations through the serving client.
@@ -129,7 +132,7 @@ func MakeFactory(ctor Ctor) rtesting.Factory {
 			return rtesting.ValidateUpdates(context.Background(), action)
 		})
 
-		actionRecorderList := rtesting.ActionRecorderList{sharedClient, dynamicClient, client, kubeClient, cachingClient, certManagerClient, routeClient}
+		actionRecorderList := rtesting.ActionRecorderList{sharedClient, dynamicClient, client, kubeClient, cachingClient, certManagerClient, routeClient, smmrClient}
 		eventList := rtesting.EventList{Recorder: eventRecorder}
 
 		return c, actionRecorderList, eventList, statsReporter
diff --git a/vendor/knative.dev/serving/pkg/reconciler/testing/v1alpha1/listers.go b/vendor/knative.dev/serving/pkg/reconciler/testing/v1alpha1/listers.go
index 4e7057b2..2212cf21 100644
--- a/vendor/knative.dev/serving/pkg/reconciler/testing/v1alpha1/listers.go
+++ b/vendor/knative.dev/serving/pkg/reconciler/testing/v1alpha1/listers.go
@@ -18,6 +18,8 @@ package v1alpha1
 
 import (
 	certmanagerv1alpha1 "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1alpha1"
+	smmrv1 "github.com/maistra/istio-operator/pkg/apis/maistra/v1"
+	smmrv1listers "github.com/openshift-knative/knative-serving-networking-openshift/pkg/client/maistra/listers/maistra/v1"
 	routev1 "github.com/openshift/api/route/v1"
 	routev1listers "github.com/openshift/client-go/route/listers/route/v1"
 	appsv1 "k8s.io/api/apps/v1"
@@ -54,6 +56,7 @@ var clientSetSchemes = []func(*runtime.Scheme) error{
 	certmanagerv1alpha1.AddToScheme,
 	autoscalingv2beta1.AddToScheme,
 	routev1.AddToScheme,
+	smmrv1.AddToScheme,
 }
 
 type Listers struct {
@@ -211,3 +214,11 @@ func (l *Listers) GetOpenshiftRouteLister() routev1listers.RouteLister {
 func (l *Listers) GetOpenshiftObjects() []runtime.Object {
 	return l.sorter.ObjectsForSchemeFunc(routev1.AddToScheme)
 }
+
+func (l *Listers) GetServiceMeshMemberRollLister() smmrv1listers.ServiceMeshMemberRollLister {
+	return smmrv1listers.NewServiceMeshMemberRollLister(l.IndexerFor(&smmrv1.ServiceMeshMemberRoll{}))
+}
+
+func (l *Listers) GetServiceMeshMemberRollObjects() []runtime.Object {
+	return l.sorter.ObjectsForSchemeFunc(smmrv1.AddToScheme)
+}
